### ISSUE TITLE
Import error on Ubuntu - undefined symbol: PyFPE_jbuf

### ISSUE THREAD

**Issue by soleilbleu** [lmcinnes/hdbscan#37]
Hi,

I have followed the instructions on Github and have successfully (apparently) installed hdbscan. However, when I import hdbscan I receive the following error:

>>> import hdbscan
Traceback (most recent call last):
File "<stdin>", line 1, in <module>
File "/home/soleilbleu/miniconda2/lib/python2.7/site-packages/hdbscan/__init__.py", line 1, in <module>
  from .hdbscan_ import HDBSCAN, hdbscan
File "/home/soleilbleu/miniconda2/lib/python2.7/site-packages/hdbscan/hdbscan_.py", line 43, in <module>
  from .plots import CondensedTree, SingleLinkageTree, MinimumSpanningTree
File "/home/soleilbleu/miniconda2/lib/python2.7/site-packages/hdbscan/plots.py", line 13, in <module>
  from sklearn.manifold import TSNE
File "/home/soleilbleu/.local/lib/python2.7/site-packages/sklearn/manifold/__init__.py", line 6, in <module>
  from .isomap import Isomap
File "/home/soleilbleu/.local/lib/python2.7/site-packages/sklearn/manifold/isomap.py", line 11, in <module>
  from ..decomposition import KernelPCA
File "/home/soleilbleu/.local/lib/python2.7/site-packages/sklearn/decomposition/__init__.py", line 11, in <module>
  from .sparse_pca import SparsePCA, MiniBatchSparsePCA
File "/home/soleilbleu/.local/lib/python2.7/site-packages/sklearn/decomposition/sparse_pca.py", line 9, in <module>
from ..linear_model import ridge_regression
File "/home/soleilbleu/.local/lib/python2.7/site-packages/sklearn/linear_model/__init__.py", line 17, in <module>
  from .coordinate_descent import (Lasso, ElasticNet, LassoCV, ElasticNetCV,
File "/home/soleilbleu/.local/lib/python2.7/site-packages/sklearn/linear_model/coordinate_descent.py", line 29, in <module>
  from . import cd_fast
ImportError: /home/soleilbleu/.local/lib/python2.7/site-packages/sklearn/linear_model/cd_fast.so: undefined symbol: PyFPE_jbuf

**Comment by lmcinnes** [comment]
This appears to be an issue in your sklearn implementation (which hdbscan depends on). You should find that a simple

    import sklearn.manifold

will produce a very similar backtrace error. I'm not sure what version of sklearn you have, but you may want to check if you can install either a newer, or older (and more stable) version. Beyond that you may want to at least verify that 'import sklearn.manifold' fails, and if so consider filing a ticket with sklearn.

**Comment by soleilbleu** [comment]
Thanks for your quick reply. The issue has been already fixed. I manually installed hdbscan using the provided scripts and also installed scikit-learn using pip rather than conda. I installed/uninstalled everything several times and the import kept giving me the same error. However, finally it was resolved by installing a specific combination of packages. I think miniconda was making some conflicts with the packages that were already installed on my Ubuntu. So I removed it, and reinstalled everything using pip or apt-get.
